pkg/i2gw/provider.go

```diff
@@ -76,7 +76,7 @@ type ResourceConverter interface {
 
 // ImplementationSpecificHTTPPathTypeMatchConverter is an option to customize the ingress implementationSpecific
 // match type conversion.
-type ImplementationSpecificHTTPPathTypeMatchConverter func(*gatewayv1.HTTPPathMatch)
+type ImplementationSpecificHTTPPathTypeMatchConverter func(*gatewayv1.HTTPPathMatch) error
```

Comment:
This seems reasonable to me, but would like to get @LiorLieberman or @mlavacca to sign off on this.

Reply:
This is to address the discussion in https://github.com/kubernetes-sigs/ingress2gateway/pull/148#discussion_r1547585006. Please let me know if it makes sense, thank you!